pandas/core/indexes/base.py

```diff
@@ -4581,7 +4582,7 @@ def _reindex_non_unique(
 
         indexer, missing = self.get_indexer_non_unique(target)
         check = indexer != -1
-        new_labels = self.take(indexer[check])
+        new_labels = self.take(indexer[check]).to_numpy()
```

Comment:
why is to_numpy necessary here?

Reply:
not necessary, it just complicates typing - I've removed it now though